agent/provider/src/cli/keystore.rs

```diff
@@ -89,3 +98,53 @@ fn remove(config: ProviderConfig, remove: Remove) -> anyhow::Result<()> {
 fn cert_dir_path(config: &ProviderConfig) -> anyhow::Result<PathBuf> {
     Ok(config.cert_dir.get_or_create()?)
 }
+
+fn print_cert_list(json_output: bool, certs_data: Vec<util::CertBasicData>) {
+    let mut table_builder = CertTableBuilder::new();
+    for data in certs_data {
+        table_builder.with_row(data);
+    }
+    let table = table_builder.build();
+    let output = CommandOutput::from(table);
+    output.print(json_output);
+}
+
+struct CertTableBuilder {
```

Comment:
Because I create same table multiple times, and builder also gives me column names I want. I could as well create tuple struct with `new` providing column names, and implement visitor for it.

Reply:
Created `CertTable` struct implementing visitor (but added extra `add` method for case it is is not used as a visitor, because `accept` would look weird.